File: 238. Product of an Array except self/PrefixPostfixArraySolution.java
class PrefixPostfixArraySolution 
{
    public int[] productExceptSelf(int[] nums) 
	{
        
        int len1 = nums.length;
        int[] prefix = new int[len1];
        int[] postfix = new int[len1];
        int[] result = new int[len1];

        for (int i = 0; i < len1; i++)
        {
            if (i == 0)
            {
                prefix[i] = 1;
            }
            else
            {
                prefix[i] = prefix[i-1]*nums[i-1];
            }
        }

        for (int i = len1-1; i > -1; i--)
        {
            if (i == len1-1)
            {
                postfix[i] = 1;
            }
            else
            {
                postfix[i] = postfix[i+1]*nums[i+1];
            }
        }

        for (int i = 0 ; i < len1 ; i ++)
        {
            result[i] = prefix[i]*postfix[i];
        }

        return result;
    }
}

File: 238. Product of an Array except self/PrefixPostfixArraySolution.py
class PrefixPostfixArraySolution:

    def productExceptSelf(self, nums: List[int]) -> List[int]:

        #Initialize a Prefix and Postfix Array
        prefix = [0]*len(nums)
        postfix = [0]*len(nums)

        #Loop through the nums array from start to end and store the prefix multiplication of each place. For 0th position the prefix will be 1.
        for i in range(len(nums)):
            if i == 0:
                prefix[i] = 1
            
            else:
                prefix[i] = prefix[i-1] * nums[i-1]

        #Loop through the nums array from end to start and store the postfix multiplication of each place. For last position the prefix will be 1.
        for i in range(len(nums)-1,-1,-1):
            if i == len(nums)-1:
                postfix[i] = 1
            
            else:
                postfix[i] = postfix[i+1] * nums[i+1]
                       
        result = [0]*len(nums)
        
        #Loop through the postfix and prefix array simultaneously and multiple the postfix and prefix multiplication of each position to get the value of product of the array except the product of that element.
        for i in range(len(nums)):
            result[i] = prefix[i]*postfix[i]

        return result

